src/OrleansProviders/Streams/Common/Monitors/IObjectPoolMonitor.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Monitor track block pool related metrics
+    /// </summary>
+    public interface IObjectPoolMonitor
+    {
+        #region Event driven metrics
+        /// <summary>
+        /// Track every time cache allocate memory from block pool
+        /// </summary>
+        /// <param name="blockCount"></param>
+        void TrackObjectAllocatedByCache(int blockCount);
```

Comment:
agree with the latter point. but I think it still make sense of keep the blockCount around for the sake of extensibility.